packages/snaps-sdk/src/ui/components/input.ts

```diff
@@ -0,0 +1,71 @@
+import type { Infer } from 'superstruct';
+import { assign, literal, object, optional, string, union } from 'superstruct';
+
+import { enumValue } from '../../internals';
+import { createBuilder } from '../builder';
+import { LiteralStruct, NodeType } from '../nodes';
+
+/**
+ * This replicates the available input types from the metamask extension.
+ * https://github.com/MetaMask/metamask-extension/develop/ui/components/component-library/input/input.constants.js
+ */
+export enum InputTypes {
```

Comment:
Enum names should be singular.

```suggestion
export enum InputType {
```

Reply:
good catch!